rectools/metrics/intersection.py

```diff
@@ -0,0 +1,130 @@
+from typing import Dict, Hashable, Optional, Union
+
+import attr
+import numpy as np
+import pandas as pd
+
+from rectools import Columns
+from rectools.metrics.base import MetricAtK
+from rectools.metrics.classification import Recall
+from rectools.utils import select_by_type
+
+
+@attr.s(auto_attribs=True)
+class Intersection(MetricAtK):
+    """
+    Metric to measure intersection in user-item pairs between recommendation lists.
+
+    The intersection@k equals the share of ``reco`` that is present in ``ref_reco``.
+
+    This corresponds to the following algorithm:
+        1) filter ``reco`` by ``k``
+        2) filter ``ref_reco`` by ``ref_k``
+        3) calculate the proportion of items in ``reco`` that are also present in ``ref_reco``
+    The second and third steps are equivalent to computing Recall@ref_k when:
+        - Interactions consists of ``reco`` without the `Columns.Rank` column.
+        - Recommendation table is ``ref_reco``
+
+    Parameters
+    ----------
+    k : int
+        Number of items in top of recommendations list that will be used to calculate metric.
+    ref_k : int, optional
+        Number of items in top of reference recommendations list that will be used to calculate metric.
+        If ``ref_k`` is None than ``ref_reco`` will be filtered with ``ref_k = k``. Default: None.
+    """
+
+    ref_k: Optional[int] = attr.ib(default=None)
+
+    def calc(self, reco: pd.DataFrame, ref_reco: pd.DataFrame) -> float:
```

Comment:
It needs to be either `ref_reco` or `ref_recos` absolutely everywhere. Each variant is ok. We can keep `ref_reco` just to make it closer to `reco` argument

Reply:
Changed